docs/tech_details/InstallationFlow.rst

```diff
@@ -0,0 +1,75 @@
+.. _app_installation_flow:
+
+App Installation Flow
+=====================
+
+Heartbeat
+---------
+
+The first thing AppAPI does is deploy of the application.
+
+In the case of ``Docker``, this is:
+
+#. 1. performing an image pull
+#. 2. creating container from the docker image
+#. 3. waiting until the “/heartbeat” endpoint becomes available with a ``GET`` request.
+
+The application, in response to the request "/heartbeat", should return json: ``{"status": "ok"}``.
+
+.. note:: The request to ``/heartbeat`` endpoint is made without AppAPI authentication.
+
+Init
+----
+
+After application is ready, which is determined by previous step,
+AppAPI sends ``POST`` request to the ``/init`` application endpoint.
+
+*Application should response with empty JSON, if initialization takes long time it should be done in background and not in this request handler.*
+
+.. note:: Starting from this point, all requests made by AppAPI contains authentications headers.
```

Comment:
Maybe here also adjust info about AppAPI auth and add ref to docs for it. And clarify which requests, to Nextcloud (AppAPI) or from it to ExApp?